src/travel-aid/TravelAidScreenComponent.tsx

```diff
@@ -48,23 +48,36 @@ import {useStopSignalMutation} from '@atb/travel-aid/use-stop-signal-mutation';
 import {MutationStatus} from '@tanstack/react-query';
 import type {SendStopSignalRequestType} from '@atb/api/stop-signal';
 import type {ContrastColor} from '@atb/theme/colors';
+import {createSentStopSignalsCache} from '@atb/travel-aid/sent-stop-signals-cache';
 
 export type TravelAidScreenParams = {
   serviceJourneyDeparture: ServiceJourneyDeparture;
 };
 type Props = TravelAidScreenParams & {
   goBack: () => void;
 };
+
+const sentStopSignalsCache = createSentStopSignalsCache();
+
 export const TravelAidScreenComponent = ({
   serviceJourneyDeparture,
   goBack,
 }: Props) => {
-  const stopSignalMutation = useStopSignalMutation();
+  const stopSignalMutation = useStopSignalMutation(() =>
+    sentStopSignalsCache.addSent(serviceJourneyDeparture),
+  );
   const styles = useStyles();
   const {t} = useTranslation();
   const {theme, themeName} = useTheme();
   const focusRef = useFocusOnLoad();
 
+  const hasSentStopSignal = sentStopSignalsCache.hasSent(
+    serviceJourneyDeparture,
+  );
+  const sendStopSignalStatus = hasSentStopSignal
+    ? 'success'
```

Comment:
Does `stopSignalMutation.status` not contain `success` as well?

Reply:
Yes it does, but if the sent stop signal is already saved, then we override/force the status value to be 'success'.